### PR TITLE
Refactoring: Use new RPR (1.34.1) material system

### PR DESCRIPTION
**Pull request by hshakula**
- Remove RPRX material system usage, use plain RPR material system instead.
- Fix numerous memory leaks due to missing 'DeleteMaterial' calls.
- Force cmake to search RPR libraries only in RPR_LOCATION_LIB

### CODE REVIEW

File: pxr/imaging/plugin/hdRpr/materialAdapter.cpp [hunk 1]
```diff
@@ -376,27 +376,27 @@ void MaterialAdapter::PoulateUsdPreviewSurface(const MaterialParams & params, co
 		}
 		else if (paramName == HdRprTokens->metallic)
 		{
-			m_vec4fRprxParams.insert({ RPRX_UBER_MATERIAL_REFLECTION_METALNESS, VtValToVec4f(paramValue)});
```
Comment (bsavery) [inline pxr/imaging/plugin/hdRpr/materialAdapter.cpp]: Is RPRX_UBER_MATERIAL_REFLECTION_METALNESS_MODE set somewhere?

**Comment by bsavery** [comment]
I think we should keep this in a branch until RPR SDK 1.34.1 is released publically.  I'll test more but looks ok on a first pass.

File: pxr/imaging/plugin/hdRpr/materialAdapter.cpp (hunk 1 above)
Comment (hshakula) [reply pxr/imaging/plugin/hdRpr/materialAdapter.cpp]: Looks like no and no one used this before
RPRX_UBER_MATERIAL_REFLECTION_METALNESS_MODE is not present in RprSupport.h and RadeonProRender.h
Are you sure that such things exist?

Comment (bsavery) [reply pxr/imaging/plugin/hdRpr/materialAdapter.cpp]: Sorry RPRX_UBER_MATERIAL_REFLECTION_MODE

Comment (hshakula) [reply pxr/imaging/plugin/hdRpr/materialAdapter.cpp]: [here](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderUSD/blob/5b12a652c648122c02a162cd2bc9a827bcbc195e/pxr/imaging/plugin/hdRpr/materialAdapter.cpp#L462) and [here](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderUSD/blob/5b12a652c648122c02a162cd2bc9a827bcbc195e/pxr/imaging/plugin/hdRpr/materialAdapter.cpp#L474)

**Pull request merged**